contract/r/gnoswap/router/v1/_helper_test.gno

```diff
@@ -511,3 +517,18 @@ func registerPositionTest(t *testing.T) {
 	testing.SetRealm(adminRealm)
 	pn.UpgradeImpl(cross, positionMockPath)
 }
+
+// mockInstanceSwapCallback is a helper function to call the SwapCallback function on the mock instance
+func mockInstanceSwapCallback(
+	token0Path string,
+	token1Path string,
+	amount0Delta string,
+	amount1Delta string,
+	payer address,
+) error {
+	// mock instance using a closure
+	return func(cur realm) error {
+		testing.SetRealm(testing.NewCodeRealm(routerPath))
```

Comment:
It looks like the realm set in the test is being ignored, and it always runs with the router realm.